src/MLIR/AST.hs

```diff
@@ -200,7 +202,13 @@ C.include "mlir-c/BuiltinTypes.h"
 C.include "mlir-c/BuiltinAttributes.h"
 
 instance FromAST Location Native.Location where
-  fromAST ctx _ UnknownLocation = Native.getUnknownLocation ctx
+  fromAST ctx _ loc = case loc of
+    UnknownLocation -> Native.getUnknownLocation ctx
+    FileLineColLocation file line col -> do
+      Native.withStringRef file \(Native.StringRef ptr len) ->
+        Native.getFileLineColLocation ctx (Native.StringRef ptr len) cline ccol
```

Comment:
There's no need to unpack the `StringRef` if you're just going to pass it along. Just do
```hs
Native.withStringRef file \fileStrRef ->
  Native.getFileLineColLocation ctx fileStrRef cline ccol
```